tests/unit/test_config.py

```diff
@@ -1,280 +1,139 @@
     """
     Confirms bool cast returns correct bools from sets of string values
     """
-    if isinstance(expected_result, bool):
-        assert globus_sdk.config._bool_cast(value) == expected_result
-    else:
-        with pytest.raises(expected_result):
-            globus_sdk.config._bool_cast(value)
+    with mock.patch.dict(os.environ):
+        os.environ["GLOBUS_SDK_HTTP_TIMEOUT"] = value
+        if expected_result is None or isinstance(expected_result, float):
+            assert globus_sdk.config.get_http_timeout() == expected_result
+        else:
+            with pytest.raises(expected_result):
+                globus_sdk.config.get_http_timeout()
 
 
-def test_get_globus_environ():
+def test_get_http_timeout_with_explicit_value():
+    with mock.patch.dict(os.environ):
+        os.environ["GLOBUS_SDK_HTTP_TIMEOUT"] = "120"
+        assert globus_sdk.config.get_http_timeout(10) == 10.0
+        assert globus_sdk.config.get_http_timeout(0) == 0.0
+        del os.environ["GLOBUS_SDK_HTTP_TIMEOUT"]
+        assert globus_sdk.config.get_http_timeout(60) == 60.0
+
+
+def test_get_environment_name():
     with mock.patch.dict(os.environ):
         # set an environment value, ensure that it's returned
         os.environ["GLOBUS_SDK_ENVIRONMENT"] = "beta"
-        assert globus_sdk.config.get_globus_environ() == "beta"
+        assert globus_sdk.config.get_environment_name() == "beta"
 
-        # clear that value, "default" should be returned
+        # clear that value, "production" should be returned
         del os.environ["GLOBUS_SDK_ENVIRONMENT"]
-        assert globus_sdk.config.get_globus_environ() == "default"
+        assert globus_sdk.config.get_environment_name() == "production"
 
         # ensure that passing a value returns that value
-        assert globus_sdk.config.get_globus_environ("beta") == "beta"
+        assert globus_sdk.config.get_environment_name("beta") == "beta"
 
 
-def test_get_globus_environ_production():
-    """
-    Confirms that get_globus_environ translates "production" to "default",
-    including when special values are passed
-    """
-    # mock environ to ensure it gets reset
-    with mock.patch.dict(os.environ):
-        os.environ["GLOBUS_SDK_ENVIRONMENT"] = "production"
-        assert globus_sdk.config.get_globus_environ() == "default"
+def test_env_config_registration():
+    with mock.patch.dict(EnvConfig._registry):
+        # should be None, we don't have an environment named 'moon'
+        assert get_config_by_name("moon") is None
 
-        del os.environ["GLOBUS_SDK_ENVIRONMENT"]
-        # ensure that passing a value returns that value
-        assert globus_sdk.config.get_globus_environ("production") == "default"
+        # now, create the moon
+        class MoonEnvConfig(EnvConfig):
+            envname = "moon"
+            domain = "apollo.globus.org"
+
+        # a lookup by "moon" should now get this config object
+        assert get_config_by_name("moon") is MoonEnvConfig
```

Comment:
`beta`, `demo`, or `sandbox2` are examples of why this might be handy.
I chose to avoid obvious names for these tests because we could actually add a "demo" or similar env in the future, and I don't want the test breaking unexpectedly.